advanced/creating-a-plugin.md

```diff
@@ -1,43 +1,113 @@
+title: Create a plugin for {{site.productname}}
 title_nav: Create a plugin
 description_short: Introducing plugin creation, with an example.
-description: A short introduction to creating plugins for TinyMCE along with an example plugin.
+description: A short introduction to creating plugins for {{site.productname}} along with an example plugin.
 keywords: plugin plugin.js plugin.min.js tinymce.js
 ---
 
-{{site.productname}} is designed to make creating plugins simple and to provide choice. Users can employ the [Yeoman generator]({{ site.baseurl }}/advanced/yeoman-generator/) or you can use the following tutorial that outlines the basic concepts of creating a plugin. Consult the API documentation and review the existing plugins that are shipped with the core editor for additional details.
+{{site.productname}} is designed to be easily extended by custom plugins.
 
-## File Structure
+## Requirements 
 
-First create a directory in the {{site.productname}} plugins directory. {{site.productname}} loads the `plugin.js` file when developers use the `tinymce.js` file in the page. Similarly, the `tinymce.min.js` loads the `plugin.min.js` file. The recommended usage is to use the `tinymce.js` file while developing and then use a build script minifying the `plugin.js` into `plugin.min.js` in production.
+To be recognised as a plugin by {{site.productname}}, the code for your custom plugin must have a JavaScript file with a single entry point that registers the plugin with {{site.productname}} using the `PluginManager`. Any other code or resources can be in separate files and can be loaded in any standard manner. {{site.productname}} also has various APIs for loading [scripts]({{ site.baseurl }}/api/tinymce.dom/tinymce.dom.scriptloader/) and [stylesheets]({{ site.baseurl }}/api/tinymce.dom/tinymce.dom).stylesheetloader/).
 
-The build scripts that comes with the {{site.productname}} development package automatically builds all plugins including custom ones.
+{{site.productname}} does not require any special file structure or tooling apart from these requirements, so custom plugins can be developed using most frameworks and tools. 
 
-## Example of the plugin file structure
+### Yeoman Generator
 
+{{site.companyname}} maintains a [Yeoman generator]({{ site.baseurl }}/advanced/yeoman-generator/) to help make plugin creation for {{site.productname}} easier. The Yeoman Generator will create the files and boilerplate code required for a custom plugin, and sets up some helpful commands. 
+
+## Registering a custom plugin with {{site.productname}}
+Register a custom plugin with {{site.productname}} using the PluginManager. `PluginManager.add()` takes a string for the plugin identifier and a function that contains the code for initializing the plugin. Optionally return an object containing a function that returns metadata which will be used in the list of plugins in the [Help plugin]({{ site.baseurl }}/plugins/opensource/help) dialog.
```

Comment:
Finding this part a bit hard to read

```
Optionally return an object containing a function that returns metadata which will be used in the list of plugins in the [Help plugin]({{ site.baseurl }}/plugins/opensource/help) dialog.
```

Reply:
Hm. I could just move this down to where I detail the metadata object...

Reply:
Yep. You're right. It should be with the metadata stuff